include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1329,45 +1346,51 @@ __parallel_reduce_by_segment_reduce_then_scan(oneapi::dpl::__internal::__device_
     assert(__n > 1);
     return __parallel_transform_reduce_then_scan(
         __backend_tag, std::forward<_ExecutionPolicy>(__exec),
-        oneapi::dpl::__ranges::make_zip_view(std::forward<_Range1>(__keys), std::forward<_Range2>(__values)),
-        oneapi::dpl::__ranges::make_zip_view(std::forward<_Range3>(__out_keys), std::forward<_Range4>(__out_values)),
+        oneapi::dpl::__ranges::zip_view(std::forward<_Range1>(__keys), std::forward<_Range2>(__values)),
```

Comment:
This change was necessary in a few places. When the range forwarding was removed in some places, `make_zip_view` resulted in a compiler error. It seemed to be caused by being unable to process l-value references to ranges. With the previous approach, we captured by value in a lambda and moved it in so this was a non-issue.

It looks like it is an instance of this issue: https://github.com/uxlfoundation/oneDPL/issues/1805.